### PR TITLE
PTCD-626 Generate Fat Export by Schedule

### PR DESCRIPTION
**Pull request by jameslawesynergy**
PTCD-626 Generate Fat Export by Schedule

### CODE REVIEW

File: dfc.providerportal.findanapprenticeship/src/dfc.providerportal.findanapprenticeship.unittests/Functions/GetApprenticeshipsAsProviderTests.cs [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+using Microsoft.Extensions.Logging.Abstractions;
+using Moq;
+using Xunit;
+
+namespace Dfc.ProviderPortal.FindAnApprenticeship.UnitTests.Functions
+{
+    public class GetApprenticeshipsAsProviderTests
+    {
+        private readonly Mock<IBlobStorageClient> _blobStorageClient;
+        private readonly Mock<Func<DateTimeOffset>> _nowUtc;
```
Comment (gunndabad) [inline dfc.providerportal.findanapprenticeship/src/dfc.providerportal.findanapprenticeship.unittests/Functions/GetApprenticeshipsAsProviderTests.cs:24]: Does this need to be a mock?

File: dfc.providerportal.findanapprenticeship/src/dfc.providerportal.findanapprenticeship.unittests/Integration/GetApprenticeshipsAsProviderIntegrationTests.cs [hunk 2]
```diff
@@ -31,8 +35,8 @@ namespace Dfc.ProviderPortal.FindAnApprenticeship.UnitTests.Integration
     public class GetApprenticeshipsAsProviderIntegrationTests
     {
         private readonly TelemetryClient _telemetryClient;
-        private readonly IAppCache _appCache;
-        private readonly Mock<IConfiguration> _configuration;
+        private readonly Mock<Func<DateTimeOffset>> _nowUtc;
```
Comment (gunndabad) [inline dfc.providerportal.findanapprenticeship/src/dfc.providerportal.findanapprenticeship.unittests/Integration/GetApprenticeshipsAsProviderIntegrationTests.cs:38]: As above

File: dfc.providerportal.findanapprenticeship/src/dfc.providerportal.findanapprenticeship/Storage/AzureBlobStorageClient.cs [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+
+        public AzureBlobStorageClient(AzureBlobStorageClientOptions options)
+        {
+            if (options == null)
+            {
+                throw new ArgumentNullException(nameof(options));
+            }
+
+            _blobContainerClient = new BlobContainerClient(options.ConnectionString, options.BlobContainerName);
+            _blobContainerClient.CreateIfNotExists(PublicAccessType.None);
```
Comment (gunndabad) [inline dfc.providerportal.findanapprenticeship/src/dfc.providerportal.findanapprenticeship/Storage/AzureBlobStorageClient.cs:19]: I'm not crazy about doing IO in constructors but if this is the simplest way then I guess it's ok

File: dfc.providerportal.findanapprenticeship/src/dfc.providerportal.findanapprenticeship.unittests/Functions/GetApprenticeshipsAsProviderTests.cs (hunk 1 above)
Comment (jameslawesynergy) [reply dfc.providerportal.findanapprenticeship/src/dfc.providerportal.findanapprenticeship.unittests/Functions/GetApprenticeshipsAsProviderTests.cs:24]: I prefer it as it makes manipulating the value returned within the tests easier.

File: dfc.providerportal.findanapprenticeship/src/dfc.providerportal.findanapprenticeship.unittests/Integration/GetApprenticeshipsAsProviderIntegrationTests.cs (hunk 2 above)
Comment (jameslawesynergy) [reply dfc.providerportal.findanapprenticeship/src/dfc.providerportal.findanapprenticeship.unittests/Integration/GetApprenticeshipsAsProviderIntegrationTests.cs:38]: I prefer it as it makes manipulating the value returned within the tests easier.

File: dfc.providerportal.findanapprenticeship/src/dfc.providerportal.findanapprenticeship/Storage/AzureBlobStorageClient.cs (hunk 3 above)
Comment (jameslawesynergy) [reply dfc.providerportal.findanapprenticeship/src/dfc.providerportal.findanapprenticeship/Storage/AzureBlobStorageClient.cs:19]: I prefer this as it limits the number of times this is executed and given it that the class' purpose is a client for the container, it seems to me like a valid dependency to enforce during construction.

**Pull request merged**